sdk/azcore/policy_bearer_token_authentication.go

```diff
@@ -0,0 +1,42 @@
+package azcore
+
+import (
+	"context"
+	"errors"
+	"time"
+)
+
+type bearerTokenAuthPolicy struct {
+	credential TokenCredential
+	refreshOn  time.Time
+	scopes     []string
+}
+
+// NewBearerTokenAuthPolicy creates a BearerTokenAuthPolicy object configured using the specified scopes and Token Credential.
+func NewBearerTokenAuthPolicy(scopes []string, cred TokenCredential) (Policy, error) {
```

Comment:
I think we should change the sig as follows.
```go
func NewBearerTokenAuthPolicy(cred TokenCredential, scopes ...string) (Policy, error)
```
Also update the comment that at least one scope must be provided.